src/index.spec.ts

```diff
@@ -79,76 +84,80 @@ describe('EppoJSClient E2E test', () => {
       const rac = readMockRacResponse();
       return res.status(200).body(JSON.stringify(rac));
     });
+    mockLogger = td.object<IAssignmentLogger>();
     globalClient = await init({
-      apiKey: 'dummy',
-      baseUrl: 'http://127.0.0.1:4000',
-      assignmentLogger: {
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        logAssignment(assignment): Promise<void> {
-          return Promise.resolve();
-        },
-      },
+      apiKey,
+      baseUrl,
+      assignmentLogger: mockLogger,
     });
   });
 
+  afterEach(() => {
+    globalClient.setLogger(mockLogger);
+    td.reset();
+  });
+
   afterAll(() => {
     mock.teardown();
   });
 
   it('assigns subject from overrides when experiment is enabled', () => {
-    const mockConfigStore = td.object<EppoLocalStorage>();
-    const mockLogger = td.object<IAssignmentLogger>();
-    td.when(mockConfigStore.get(hashedFlagKey)).thenReturn({
-      ...mockExperimentConfig,
-      overrides: {
-        '1b50f33aef8f681a13f623963da967ed': 'variant-2',
-      },
-      typedOverrides: {
-        '1b50f33aef8f681a13f623963da967ed': 'variant-2',
-      },
+    td.replace(EppoLocalStorage.prototype, 'get', (key: string) => {
```

Comment:
Since we're not passing the configuration store in `init()`, which is what I want to test in this file, instead of creating a mock store that we pass we instead replace the store's `get()` function.